internal/backends/clojure/deps.go

```diff
@@ -0,0 +1,504 @@
+	if len(pkgPom.Developers.Developer) > 0 {
+		author = pkgPom.Developers.Developer[0].Name
+	}
+	return MPkgInfo{
+		Id:            pkg,
+		LatestVersion: metadata.Versioning.LatestVersion,
+		Versions:      versions,
+		Description:   pkgPom.Description,
+		ScmUrl:        pkgPom.Scm.Url,
+		Author:        author,
+	}, nil
+}
+
+func GetMPackageInfo(pkg JavaPackage) (MPkgInfo, error) {
+	minfo, err := GetMPackageInfoMaven(pkg)
+	if err == nil {
+		return minfo, nil
+	} else {
+		cinfo, err := GetMPackageInfoClojars(pkg)
+		if err == nil {
+			return cinfo, nil
+		}
+		return MPkgInfo{}, errors.New("Couldn't fetch info for package: " + pkg.toString() + " from any sources")
+	}
+}
+
+func GenPkgInfo(pinfo MPkgInfo) api.PkgInfo {
+	return api.PkgInfo{
+		Name:             pinfo.Id.Group + "/" + pinfo.Id.Artifact,
+		Description:      pinfo.Description,
+		Version:          pinfo.LatestVersion,
+		HomepageURL:      pinfo.Homepage,
+		DocumentationURL: "",
+		SourceCodeURL:    "",
+		BugTrackerURL:    "",
+		Author:           pinfo.Author,
+		License:          "",
+		Dependencies:     nil,
+	}
+}
+
+func GetPackageInfo(pkg JavaPackage) (api.PkgInfo, error) {
+	pinfo, err := GetMPackageInfo(pkg)
+	if err != nil {
+		return api.PkgInfo{}, err
+	}
+	return GenPkgInfo(pinfo), nil
+}
+
+func findPackagesClojars(name string) ([]api.PkgInfo, error) {
+	s := "https://clojars.org/search?q=" + name + "&format=json"
+	res, err := http.Get(s)
+	if err != nil || res.StatusCode != 200 {
+		return nil, errors.New("Couldn't fetch package info")
+	}
+	var resp ClojarsSearchOp
+	bs, err := ioutil.ReadAll(res.Body)
+	err = json.Unmarshal(bs, &resp)
+	if err != nil {
+		return nil, err
+	}
+
+	fres := []api.PkgInfo{}
+	for _, p := range resp.Results {
+		pinfo, err := GetMPackageInfoClojars(JavaPackage{
+			Artifact: p.JarName,
+			Group:    p.GroupName,
+		})
+		if err != nil {
+			panic(err)
+		}
+		fres = append(fres, GenPkgInfo(pinfo))
+	}
+
+	return fres, nil
+}
+
+func findPackagesMaven(name string) ([]api.PkgInfo, error) {
+	s := "https://search.maven.org/solrsearch/select?q=\"" + name + "\"&rows=20&wt=json"
+	res, err := http.Get(s)
+	if err != nil || res.StatusCode != 200 {
+		return nil, errors.New("Couldn't fetch package info")
+	}
+	var resp MavenSearchOp
+	bs, err := ioutil.ReadAll(res.Body)
+	err = json.Unmarshal(bs, &resp)
+	if err != nil {
+		return nil, err
+	}
+
+	fres := []api.PkgInfo{}
+	for _, p := range resp.Response.Docs {
+		pinfo, err := GetMPackageInfoMaven(JavaPackage{
+			Artifact: p.Artifact,
+			Group:    p.Group,
+		})
+		if err != nil {
+			panic(err)
+		}
+		fres = append(fres, GenPkgInfo(pinfo))
+	}
+
+	return fres, nil
+}
+
+func Search(name string) []api.PkgInfo {
+	cpkg, err := findPackagesClojars(name)
+	if err != nil {
+		util.Log("Error while querying clojars: " + err.Error())
+	}
+	mpkg, err := findPackagesMaven(name)
+	if err != nil {
+		util.Log("Error while querying maven: " + err.Error())
+	}
+	return append(mpkg, cpkg...)
```

Comment:
Is it possible for a single package to come back from clojars and from maven? If so, we'll want to remove duplicates.

Reply:
Yes, I'll get it done.